File: src/main/java/de/thatsich/intellie/common/registries/ItemRegistry.java
package de.thatsich.intellie.common.registries;

import cpw.mods.fml.common.registry.GameRegistry;
import de.thatsich.intellie.common.module.item.IItem;
import de.thatsich.intellie.common.util.logging.ILog;
import net.minecraft.item.Item;

import javax.inject.Inject;
import javax.inject.Singleton;
import java.util.Collection;
import java.util.LinkedList;

/**
 Handles registration of items

 @author thatsIch
 @date 25.02.14. */
@Singleton
public class ItemRegistry implements IRegistry
{
	private final ILog log;
	private final Collection<IItem> items;

	/**
	 @param log Log
	 */
	@Inject
	ItemRegistry ( final ILog log )
	{
		this.log = log;
		this.items = new LinkedList<IItem>();
	}

	public void addItem ( IItem item )
	{
		this.items.add( item );
		this.log.debug( "Added Item %s.", item );
	}

	@Override
	public void register ()
	{
		for ( final IItem item : this.items )
		{
			final String unlocalizedName = item.getUnlocalizedName();
			GameRegistry.registerItem( (Item) item, unlocalizedName );
			this.log.debug( "Registered Item %s with %s", item, unlocalizedName );
		}
		this.log.info( "Registering all Items." );
	}
}

File: src/main/scala/de/thatsich/minecraft/core/registries/ItemRegistry.scala
package de.thatsich.minecraft.core.registries

import de.thatsich.minecraft.core.module.item.IItem
import cpw.mods.fml.common.registry.GameRegistry
import net.minecraft.item.Item
import de.thatsich.minecraft.core.log.ILog

/**
 *
 *
 * @author thatsIch
 * @since 06.04.2014.
 */
final class ItemRegistry(implicit log: ILog)
	extends ARegistry[ IItem ](log)
{
	def register()
	{
		for( item <- this.set )
		{
			val unlocalizedName = item.getUnlocalizedName

			GameRegistry.registerItem(item.asInstanceOf[ Item ], unlocalizedName)
			this.log.debug("Registered Item %s with %s", item, unlocalizedName)
		}
		this.log.info("Registering all Items.")
	}
}
